### PR TITLE
operator: Include runtime-config in compiling the SHA1 checksum

### PR DESCRIPTION
**Pull request by btaani**
**What this PR does / why we need it**:
Any changes made to the runtime-config can now trigger the reconciliation of the LokiStack.

**Which issue(s) this PR fixes**:
Fixes:
[LOG-3953](https://issues.redhat.com/browse/LOG-3953) - Updating the RulerConfig CR can now trigger the restart of the ruler pod (and thus reflecting the changes to the user-workload AM)
[LOG-3950](https://issues.redhat.com/browse/LOG-3950) - Enabling user-workload AM

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [x] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

**Review by periklis — CHANGES_REQUESTED** [review]
The proposed change set works against our first principles that the operator should delegate anything possible to the Kubernetes APIServer instead of inventing the wheel anew. 

By saying this, I suggest you take a look on the code path starting from this snippet to discover how we instruct k8s to restart pods by passing annotations to the pod template spec:
https://github.com/grafana/loki/blob/11f5218ebf1eefc2b34355b55b5a468c7c4ad0ce/operator/internal/handlers/lokistack_create_or_update.go#L267-L270

I.e. The ruler config annotation is something that is updated on the LokiStack CR on every RulerConfig change. Affected components, e.g. ruler needs this annotation to be passed to the pod template spec.

**Comment by btaani** [comment]
@periklis updating the annotations is already implemented [here](https://github.com/grafana/loki/blob/f67c529ba2615ae308bf941e13787d407a928d9d/operator/controllers/loki/internal/lokistack/ruler_config_discovery.go#L33) for RulerConfig changes. However, updating anything below the `overrides` section of the RulerConfig CR updates the `annotationRulerConfigDiscoveredAt` value for the lokistack, but doesn't seem to trigger restarting the ruler pod, which is why I explicitly restart it like this.

File: operator/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 - [9511](https://github.com/grafana/loki/pull/9511) **xperimental**: Do not update status after setting degraded condition
 - [9405](https://github.com/grafana/loki/pull/9405) **periklis**: Add support for configuring HTTP server timeouts
+- [9448](https://github.com/grafana/loki/pull/9448) **btaani**: Include runtime-config in compiling the SHA1 checksum
```
Comment (periklis) [inline operator/CHANGELOG.md:5]:
```suggestion
- [9448](https://github.com/grafana/loki/pull/9448) **btaani**: Include runtime-config in compiling the SHA1 checksum
- [9511](https://github.com/grafana/loki/pull/9511) **xperimental**: Do not update status after setting degraded condition
- [9405](https://github.com/grafana/loki/pull/9405) **periklis**: Add support for configuring HTTP server timeouts
```

File: operator/controllers/loki/internal/lokistack/ruler_config_discovery.go [hunk 2]
```diff
@@ -5,10 +5,11 @@ import (
 	"time"
 
 	"github.com/ViaQ/logerr/v2/kverrors"
-	lokiv1 "github.com/grafana/loki/operator/apis/loki/v1"
-	"github.com/grafana/loki/operator/internal/external/k8s"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	lokiv1 "github.com/grafana/loki/operator/apis/loki/v1"
+	"github.com/grafana/loki/operator/internal/external/k8s"
```
Comment (periklis) [inline operator/controllers/loki/internal/lokistack/ruler_config_discovery.go:12]: Do we need this change?

File: operator/controllers/loki/internal/lokistack/update.go [hunk 3]
```diff
@@ -3,11 +3,12 @@ package lokistack
...
 	"context"
 
-	lokiv1 "github.com/grafana/loki/operator/apis/loki/v1"
-	"github.com/grafana/loki/operator/internal/external/k8s"
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/client-go/util/retry"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	lokiv1 "github.com/grafana/loki/operator/apis/loki/v1"
+	"github.com/grafana/loki/operator/internal/external/k8s"
```
Comment (periklis) [inline operator/controllers/loki/internal/lokistack/update.go:11]: Do we need this change?

File: operator/controllers/loki/internal/lokistack/ruler_config_discovery.go (hunk 2 above)
Comment (btaani) [reply operator/controllers/loki/internal/lokistack/ruler_config_discovery.go:12]: No we don't. This was from the previous commits

File: operator/CHANGELOG.md (hunk 1 above)
Comment (periklis) [inline operator/CHANGELOG.md:5]:
```suggestion
- [9448](https://github.com/grafana/loki/pull/9448) **btaani**: Include runtime-config in compiling the SHA1 checksum
- [9511](https://github.com/grafana/loki/pull/9511) **xperimental**: Do not update status after setting degraded condition
-
 [9405](https://github.com/grafana/loki/pull/9405) **periklis**: Add support for configuring HTTP server timeouts
```

File: operator/CHANGELOG.md [hunk 4]
```diff
@@ -1,7 +1,9 @@
 ## Main
 
+- [9448](https://github.com/grafana/loki/pull/9448) **btaani**: Include runtime-config in compiling the SHA1 checksum
 - [9511](https://github.com/grafana/loki/pull/9511) **xperimental**: Do not update status after setting degraded condition
-- [9405](https://github.com/grafana/loki/pull/9405) **periklis**: Add support for configuring HTTP server timeouts
+-
+ [9405](https://github.com/grafana/loki/pull/9405) **periklis**: Add support for configuring HTTP server timeouts
```
Comment (periklis) [inline operator/CHANGELOG.md:6]:
```suggestion
- [9405](https://github.com/grafana/loki/pull/9405) **periklis**: Add support for configuring HTTP server timeouts
```

**Review by periklis — APPROVED** [review]

**Pull request merged**